tsdb/index/index.go

```diff
@@ -959,14 +982,12 @@ func (w *Writer) writePosting(name, value string, offs []uint32) error {
 	w.cntPO++
 
 	w.buf1.Reset()
-	w.buf1.PutBE32int(len(offs))
-
 	for _, off := range offs {
 		if off > (1<<32)-1 {
```

Comment:
No idea, my guess is that someone forgot to upgrade this function to use `storage.SeriesRef`?

Reply:
I don't have enough context.  Should `PostingsEncoder` take `[]storage.SeriesRef` ?
I agree this could be done in another PR, but I am (slightly) concerned that this PR turns one loop into two loops, when one of them is pointless.

That is why I thought about having the encoder return an error, then you could move the check too.